### ISSUE TITLE
[Feature]: opts for visual and normal modes?

### ISSUE THREAD

**Issue by olimorris** [mrjones2014/legendary.nvim#105]
### Similar Issues

- [X] Before filing, I have searched for similar issues.

### Description

With the brilliant new feature added yesterday, enabling us to do this:

```lua
{ '<LocalLeader>"', { n = [[ciw"<c-r>""<esc>]], v = [[c"<c-r>""<esc>]] }, description = "Wrap in quotes" },
```

I quickly began trying to consolidate my keymaps. I have a very advanced multiple cursors mapping based on this excellent [article](http://www.kevinli.co/posts/2017-01-19-multiple-cursors-in-500-bytes-of-vimscript):

```lua
vim.g.mc = vim.api.nvim_replace_termcodes([[y/\V<C-r>=escape(@", '/')<CR><CR>]], true, true, true)

{ "cn", "*``cgN", description = "Multiple cursors (backwards)" }
{ "cn", [[g:mc . "``cgn"]], description = "Multiple cursors", mode = { "x" }, opts = { expr = true } }
```

Unfortunately I can't consolidate them with the new feature as `expr = true` needs to be present for the `x` mode and *not* for the `n` mode.

If this is too much work, then forget it. It just saves a line and an extra row in my legendary popup. Thought it may be an interesting feature though.

**Comment by mrjones2014** [comment]
Shouldn’t be too much work at all.

my one concern (not really a concern even, more of a question), what do we do with outer opts if inner opts also exist, merge or overwrite?

**Comment by mrjones2014** [comment]
Also how the heck do you do multi cursors? I've been trying to figure that out for ages lol

**Comment by mrjones2014** [comment]
So I see two possible APIs for this:

```lua
{
   '<LocalLeader>"',
  {
    n = { [[ciw"<c-r>""<esc>]], opts = { }  },
    v = { [[c"<c-r>""<esc>]], opts = { } }
  },
  description = "Wrap in quotes"
}
```

or:

```lua
{
   '<LocalLeader>"',
  {
    n = [[ciw"<c-r>""<esc>]],
    v = [[c"<c-r>""<esc>]], 
  },
  description = "Wrap in quotes",
  opts = {
    n = {},
    v = {}
  }
}
```

I think I'm leaning towards the first as it'll be easier to parse. But the question with that one is, if outer and inner opts both exist, do we merge or overwrite, and which one takes precedence in either case? My guy says "merge" and "inner takes precedence" but I'd like to hear your thoughts.

**Comment by olimorris** [comment]
> So I see two possible APIs for this:
> 
> ```lua
> {
>    '<LocalLeader>"',
>   {
>     n = { [[ciw"<c-r>""<esc>]], opts = { }  },
>     v = { [[c"<c-r>""<esc>]], opts = { } }
>   },
>   description = "Wrap in quotes"
> }
> ```

This is really nice! Feels super clean. Easy to do?

> Also how the heck do you do multi cursors? I've been trying to figure that out for ages lol
>
>EDIT: ah I see you've linked the article

Also worth seeing my neovim translated mappings [here](https://github.com/olimorris/dotfiles/blob/main/.config/nvim/lua/Oli/core/mappings.lua#L79-L121). The original ones from the article don't work in Neovim/with Lua. But the videos are good to practice with. The multi-cursor macros are just insane. I had to track the guy down on LinkedIn just to thank him 😉 .

**Comment by mrjones2014** [comment]
Yup that first syntax shouldn't be too difficult. I'm thinking "merge" and "inner takes precedence" for `opts`.

Akinsho makes great plugins, I've chatted with him in several issues.
